### PR TITLE
Link cases and promises

### PR DESCRIPTION
**Pull request by AriHrannar**
## Status
**Ready**

## Description
Cronjob that connects cases and promises
They get a 'Suggested' type if they share 50%-79% of the subjects
They get a 'Connected' type if they share 80% or more of the subjects

## TODO
- [x] Fix error when PromiseCase already exists in DB (should not create duplicate connections)

### CODE REVIEW

File: src/promises/services.py [hunk 1]
```diff
@@ -75,5 +75,5 @@ def find_connected_bills_and_promises():
                             CRONLOGGER.info('PromiseCase connection created')
                         # If the percent is between 50% and 79% we create a suggested connection
                         elif percent_of_common_subjects >= 0.5:
-                            SuggestedPromiseCase.objects.create(case=case, promise=promise)
+                            # SuggestedPromiseCase.objects.create(case=case, promise=promise)
```
Comment (Ragnar-H) [inline src/promises/services.py]: Remove the call entirely maybe?

File: src/promises/cron_jobs.py [hunk 2]
```diff
@@ -8,17 +8,18 @@
...
-    code = 'promise.connect_bills_and_promises'
+    code = 'promises.cron_jobs.connect_cases_and_promises'
 
 
     def do(self):
         ''' Executes the cron job '''
         try:
             CRONLOGGER.info('Starting bill and promise connection')
             promise_services.find_connected_bills_and_promises()
+            CRONLOGGER.info('Completed bill and promise connection')
```
Comment (Ragnar-H) [inline src/promises/cron_jobs.py]: \o/

File: src/promises/services.py [hunk 3]
```diff
@@ -1,79 +1,52 @@
...
+    cases_in_sessions = Case.objects.filter(parliament_session__in=parliament_sessions_ids)
+
+    for case in cases_in_sessions:
+        case_subject_ids = CaseSubject.objects.filter(case=case).values_list('subject_id', flat=True)
+        
+        # We look at each promise within the same parliament
+        for promise in current_promises:
+            promise_subject_ids = PromiseSubject.objects.filter(promise=promise).values_list('subject_id', flat=True)
+            if len(promise_subject_ids) > 4:
+                # We want more than 4 subjects on the promise
```
Comment (Ragnar-H) [inline src/promises/services.py]: Maybe a small note on _why_ 5 is the lowest number?
Even though it's probably subject to change

File: src/promises/services.py [hunk 4]
```diff
@@ -1,79 +1,52 @@
...
+                
+                # If the case and promise share a certain percentage of subjects
+                # then we create a connection between them
+                if percent_of_common_subjects >= 50:
+                    relationship_type = 'Suggested'
+                    if percent_of_common_subjects >= 80:
+                        relationship_type = 'Connected'
+                    
+                    # We check if the connection already exists
+                    casePromiseConnection = PromiseCase.objects.filter(case=case, promise=promise)
```
Comment (Ragnar-H) [inline src/promises/services.py]: huh, I'd think that Django had some sort of utility function for this

Comment (AriHrannar) [reply src/promises/services.py]: there is the exists() function, but if it exists I still want to update a field, so its easier to move that into a variable, check if that exists and then update the field & save

**Pull request merged**